bot.py

```diff
@@ -89,32 +97,78 @@ async def on_member_join(member):
     name="help",
     help="Displays the syntax and the description of all the commands.",
 )
-async def help_command(ctx):
+async def help_command(ctx, name=None):
     """Help command outputs description about all the commands"""
     try:
+        # typing indicator
+        async with ctx.typing():
+            await asyncio.sleep(1)
+
+        list_commands = []  # make dictionary to store all commands
+        for command in bot.commands:
+            # check if the user input specific command after help
+            if command.name == name:
+                # for commands that receive multiple parameter
+                if command.name in ["add", "remove"]:
```

Comment:
What is this doing??
Why are we looping for commands and handle certain commands in special way?

Reply:
because `add` and `remove` allow two or more summoners like `add_summoner(ctx, *, message="--help")`.
However other commands allow only one summoner or nothing like `get_rank(ctx, name="--help")` or `display_teams(ctx, name=None)`

`add` and `remove` were specially checked cuz the number of parameters is different from others.

Reply:
hmm okay I really don't like this - see if you make this simpler by refactoring other places too.

Reply:
okay. I will do my BEST !!